### PR TITLE
Kotlin 1.8.21 updates

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

**Comment by ZacSweers** [comment]
https://youtrack.jetbrains.com/issue/KT-56645

**Comment by ZacSweers** [comment]
Should be fixed in 1.8.20-RC

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 [versions]
 coroutines = "1.6.4"
-kotlin = "1.8.0"
+kotlin = "1.8.20-RC"
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:3]:
```suggestion
kotlin = "1.8.21"
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 jvmTarget = "11"
+ksp = "1.8.20-RC-1.0.9"
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:5]:
```suggestion
ksp = "1.8.21-1.0.11"
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -11,9 +12,9 @@ retrofit = "2.9.0"
 binaryCompatibilityValidator = { id = "org.jetbrains.kotlinx.binary-compatibility-validator", version = "0.11.1" }
 detekt = { id = "io.gitlab.arturbosch.detekt", version = "1.21.0" }
 dokka = { id = "org.jetbrains.dokka", version = "1.7.20" }
-ksp = { id = "com.google.devtools.ksp", version = "1.8.0-1.0.8" }
-mavenPublish = { id = "com.vanniktech.maven.publish", version = "0.23.1" }
-spotless = { id = "com.diffplug.spotless", version = "6.12.0" }
+ksp = { id = "com.google.devtools.ksp", version.ref = "ksp" }
+mavenPublish = { id = "com.vanniktech.maven.publish", version = "0.24.0" }
+spotless = { id = "com.diffplug.spotless", version = "6.15.0" }
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:17]:
```suggestion
spotless = { id = "com.diffplug.spotless", version = "6.18.0" }
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -11,9 +12,9 @@ retrofit = "2.9.0"
 binaryCompatibilityValidator = { id = "org.jetbrains.kotlinx.binary-compatibility-validator", version = "0.11.1" }
 detekt = { id = "io.gitlab.arturbosch.detekt", version = "1.21.0" }
 dokka = { id = "org.jetbrains.dokka", version = "1.7.20" }
-ksp = { id = "com.google.devtools.ksp", version = "1.8.0-1.0.8" }
-mavenPublish = { id = "com.vanniktech.maven.publish", version = "0.23.1" }
-spotless = { id = "com.diffplug.spotless", version = "6.12.0" }
+ksp = { id = "com.google.devtools.ksp", version.ref = "ksp" }
+mavenPublish = { id = "com.vanniktech.maven.publish", version = "0.24.0" }
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:16]:
```suggestion
mavenPublish = { id = "com.vanniktech.maven.publish", version = "0.25.2" }
```

**Review by ZacSweers — COMMENTED** [review]
Update more

**Review by MohamedAbdalkader — APPROVED** [review]

**Pull request merged**